scipy/optimize/__init__.py

```diff
@@ -230,6 +230,7 @@
 
 from __future__ import division, print_function, absolute_import
 
+from .constraints import *
```

Comment:
Looks like both ``constraints`` and ``minimize_constrained`` should be renamed to start with an underscore (they're not public submodules).